amy/templates/dashboard/training_progress.html

```diff
@@ -71,22 +109,39 @@ <h2>Your progress of becoming The Carpentries Instructor</h2>
     <tr>
       <th>3. Welcome Session</th>
       <td>
-        {% if user.passed_welcome %}
-          <p class="text-success">Welcome Session passed.</p>
+        {% if progress_welcome %}
+          {% progress_trainee_view progress_welcome %}
+        {% elif user.passed_welcome %}
+          <p class="text-success">Welcome Session completed.
+          </p>
         {% else %}
-          <p>Welcome Session not passed yet.</p>
-          <p>Register for a Welcome Session on <a href="https://pad.carpentries.org/welcome-sessions-{% now 'Y' %}">this Etherpad</a>.</p>
+          <p>Welcome Session not completed yet.</p>
+          {% now 'n' as current_month %}
+          {% now 'Y' as current_year %}
+          <p>Register for a Welcome Session on
+            {% if current_month|add:"0" >= 11 %}
+            one of these Etherpads: <a href="https://pad.carpentries.org/welcome-sessions-{{ current_year }}">Welcome Sessions {{ current_year }}</a>; <a href="https://pad.carpentries.org/welcome-sessions-{{ current_year | add:'1' }}">Welcome Sessions {{ current_year | add:"1" }}</a>.
```

Comment:
Calculating proper etherpad URL seems so complex it could become a template tag.

Reply:
Switched to a template tag. Conveniently that makes it more testable :)